src/napari_spatialdata/_view.py

```diff
@@ -19,12 +19,203 @@
 from napari_spatialdata._widgets import (
     CBarWidget,
     AListWidget,
+    ComponentWidget,
     ObsmIndexWidget,
+    MatplotlibWidget,
     RangeSliderWidget,
+    ScatterListWidget,
 )
 from napari_spatialdata._constants._pkg_constants import Key
 
-__all__ = ["QtAdataViewWidget"]
+__all__ = ["QtAdataViewWidget", "QtAdataScatterWidget"]
+
+
+class QtAdataScatterWidget(QWidget):
+    """Adata viewer widget."""
+
+    def __init__(self, viewer: Viewer):
+        super().__init__()
+
+        self._viewer = viewer
+        self._model = ImageModel()
+
+        self._layer_selection_widget = magicgui(
+            self._select_layer,
+            layer={"choices": self._get_layer},
+            auto_call=True,
+            call_button=False,
+        )
+        self._layer_selection_widget()
+
+        self.setLayout(QVBoxLayout())
+        self.layout().addWidget(self._layer_selection_widget.native)
+
+        # Scrollbar
+
+        # Matplotlib
+
+        self.matplotlib_widget = MatplotlibWidget(self.viewer, self.model)
+        self.layout().addWidget(self.matplotlib_widget)
+
+        # Dropdown menu to select between obs, obsm, var for X axis
+        x_selection_label = QLabel("Select type for X axis:")
+        x_selection_label.setToolTip("Select between obs, obsm and var.")
+        self.x_selection_widget = QComboBox()
```

Comment:
we discussed about ordering, is QComboBox still the preferred solution or is there an alterantive to use space better in the widget?

Reply:
We switched from QVBoxLayout to QGridLayout -> the UI looks better now.